catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswEndpoint.java

```diff
@@ -662,29 +669,34 @@ private int deleteRecords(DeleteAction deleteAction)
         queryRequest = queryFactory.updateQueryRequestTags(queryRequest,
                 schemaTransformerManager.getTransformerSchemaForId(deleteAction.getTypeName()));
 
-        QueryResponse response = framework.query(queryRequest);
-
-        List<String> ids = new ArrayList<>();
-
-        for (Result result : response.getResults()) {
-            if (result != null && result.getMetacard() != null) {
-                ids.add(result.getMetacard()
-                        .getId());
+        Iterable<List<Result>> resultLists = Iterables.partition(ResultIterable.resultIterable(
+                framework,
+                queryRequest), DEFAULT_BATCH);
+        AtomicInteger batch = new AtomicInteger(1);
```

Comment:
Mostly just consistency. I need the in-out objects in the `updateRecords()` method that is similar to this. I can change them here if it bothers you.